### ISSUE TITLE
Example from Documentation doesn't work

### ISSUE THREAD

**Issue by Will-So** [TkTech/pysimdjson#90]
When installing pysimdjson 4.0.2 and running the example from the documentation:

```python
import simdjson
parser = simdjson.Parser()

for i in range(0, 100):
    doc = parser.parse(b'{"a": "b"}')
```

I get the following error:

```
RuntimeError: Tried to re-use a parser while simdjson.Object and/or simdjson.Array objects still exist referencing the old parser.
```

This seems to be related to this:

https://github.com/TkTech/pysimdjson/issues/82

Is the most performant way to solve this to `del doc` at each stage in the loop? If so I can add that to the docs.

**Comment by TkTech** [comment]
Oof you're right, that's bad. The "correct" fix there is `del doc` at the end of the loop, but I really don't like manual memory management like this. #82 was a quick fix to prevent users from shooting themselves in the foot, but there has to be a better way.

**Comment by TkTech** [comment]
"Fixed" as of version 4.0.3.
